native/packages/react-native-bpk-component-badge/src/BpkBadge.js

```diff
@@ -0,0 +1,179 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { setOpacity } from 'bpk-tokens';
+import BpkText from 'react-native-bpk-component-text';
+import {
+  colorGray700,
+  colorGray50,
+  colorYellow500,
+  colorRed500,
+  colorGreen500,
+  colorWhite,
+  borderRadiusSm,
+  borderSizeSm,
+  spacingSm,
+  spacingMd,
+} from 'bpk-tokens/tokens/base.react.native';
+import { View, ViewPropTypes, StyleSheet } from 'react-native';
+
+const styles = StyleSheet.create({
+  viewBase: {
+    paddingHorizontal: spacingMd,
+    paddingVertical: spacingSm,
+  },
+  viewWarning: {
+    backgroundColor: colorYellow500,
+  },
+  viewSuccess: {
+    backgroundColor: colorGreen500,
+  },
+  viewDestructive: {
+    backgroundColor: colorRed500,
+  },
+  viewLight: {
+    backgroundColor: colorGray50,
+  },
+  viewInverse: {
+    backgroundColor: colorWhite,
+  },
+  viewOutline: {
+    paddingHorizontal: spacingMd - borderSizeSm,
+    paddingVertical: spacingSm - borderSizeSm,
+    backgroundColor: setOpacity(colorWhite, 0.2),
+  },
+  textBase: {
+    color: colorGray700,
+  },
+  textDestructive: {
+    color: colorWhite,
+  },
+  textOutline: {
+    color: colorWhite,
+  },
+  borderBase: {
+    borderTopLeftRadius: borderRadiusSm,
+    borderBottomLeftRadius: borderRadiusSm,
+    borderTopRightRadius: borderRadiusSm,
+    borderBottomRightRadius: borderRadiusSm,
+  },
+  borderOutline: {
+    borderWidth: borderSizeSm,
+    borderColor: colorWhite,
+  },
+  borderLeft: {
+    borderTopLeftRadius: 0,
+    borderBottomLeftRadius: 0,
+    borderTopRightRadius: 0,
+    borderBottomRightRadius: borderRadiusSm,
+  },
+  borderRight: {
+    borderTopLeftRadius: 0,
+    borderBottomLeftRadius: borderRadiusSm,
+    borderTopRightRadius: 0,
+    borderBottomRightRadius: 0,
+  },
+});
+
+export const BADGE_TYPES = {
+  success: 'success',
+  warning: 'warning',
+  destructive: 'destructive',
+  light: 'light',
+  inverse: 'inverse',
+  outline: 'outline',
+};
+
+export const BADGE_DOCKED_TYPES = {
+  left: 'left',
+  right: 'right',
+};
+
+export type Props = {
+  message: string,
+  docked: $Keys<typeof BADGE_DOCKED_TYPES>,
+  type: $Keys<typeof BADGE_TYPES>,
+  style: ?(Object | Array<Object>),
+};
+
+const viewStyleMap = {
+  [BADGE_TYPES.success]: styles.viewSuccess,
+  [BADGE_TYPES.warning]: styles.viewWarning,
+  [BADGE_TYPES.destructive]: styles.viewDestructive,
+  [BADGE_TYPES.light]: styles.viewLight,
+  [BADGE_TYPES.inverse]: styles.viewInverse,
+  [BADGE_TYPES.outline]: [styles.viewOutline, styles.borderOutline],
+};
+
+const textStyleMap = {
+  [BADGE_TYPES.success]: null,
```

Comment:
Im not sure what people think of this... this was to fix the flow warning. I think it' makes it quite explicit that there are no text styles that are specific to `success`, `warning`, `light` or `inverse`. Alternatively i could check for existence and then push the style on line 143

Reply:
Why doesn't Flow like it without this? Is it because the value might be undefined and you cannot push undefined? If so it'd be better to write it with a conditional on line 143 imo

Reply:
Fixed with type annotation: `{ [key: string]: Object | Array<Object> }`